### PR TITLE
Log when FunctionDispatcher is initialized.

### LINKED ISSUE #4384: Add log to signify when debugger can attach
In the VS Code extension, we look for `Host lock lease acquired by instance ID` to signal that we can attach the debugger. We originally used `Job host started` but that apparently didn't work in all cases - see https://github.com/Microsoft/vscode-azurefunctions/issues/660.

However, the "host lock lease" message doesn't always get printed - see [here](https://github.com/Azure/azure-functions-host/blob/2dd52a0eaf29d5d84787d2c367867532f3c36e66/src/WebJobs.Script/Host/PrimaryHostCoordinator.cs#L141). We ran into this when trying to debug with Azurite, which doesn't seem to work well with leases - see [here](https://github.com/Microsoft/vscode-azurefunctions/issues/1245). A few other concerns:
1. I've also always been a little nervous that someone will change the "Host lock lease" message without realizing that it will break VS Code debugging
1. The message feels like an internal implementation detail that shouldn't really matter for debugging

Can we add a message that's sole purpose is to signify that debuggers can attach? I think the problem with `Job host started` in https://github.com/Microsoft/vscode-azurefunctions/issues/660 was that the _worker_ wasn't fully started yet. Perhaps we just need a message like `Language worker started`? I honestly don't really care about the wording as long as it meets these criteria:
1. It always gets printed, regardless of lease status
1. It's unique enough that we won't run into false positives from other logs
1. We agree that the message won't change going forward

### PR DESCRIPTION
**Pull request by pragnagopa**
Fixes #4384

### CODE REVIEW

**Comment by pragnagopa** [comment]
cc  @EricJizbaMSFT 

verrfied debugger breaks into a js function with runonstartup set to true for the first invocation with following tasks.json

```json
{
  "version": "2.0.0",
  "tasks": [
    {
      "type": "func",
      "command": "host start",
      "isBackground": true,
      "dependsOn": "npm install",
       "problemMatcher": {
        "base": "$func-watch",
        "background": {
          "activeOnStart": true,
          "beginsPattern": "^.*(Job host stopped|signaling restart).*$",
          "endsPattern": "^.*Worker process started and initialized.*$"
        },
       }
    },
    {
      "type": "shell",
      "label": "npm install",
      "command": "npm install"
    },
    {
      "type": "shell",
      "label": "npm prune",
      "command": "npm prune --production",
      "problemMatcher": []
    }
  ]
}

```

File: src/WebJobs.Script/Workers/Http/HttpFunctionInvocationDispatcher.cs [hunk 1]
```diff
@@ -65,15 +65,14 @@ internal class HttpFunctionInvocationDispatcher : IFunctionInvocationDispatcher
...
         {
-            // TODO: Add process managment for http invoker
-
             _httpWorkerChannel = _httpWorkerChannelFactory.Create(_scriptOptions.RootScriptPath, _metricsLogger, attemptCount);
             _httpWorkerChannel.StartWorkerProcessAsync(cancellationToken).ContinueWith(workerInitTask =>
                  {
                      if (workerInitTask.IsCompleted)
                      {
                          _logger.LogDebug("Adding http worker channel. workerId:{id}", _httpWorkerChannel.Id);
                          State = FunctionInvocationDispatcherState.Initialized;
```
Comment (mhoeger) [inline src/WebJobs.Script/Workers/Http/HttpFunctionInvocationDispatcher.cs]: Nit: remove this line?

File: src/WebJobs.Script/Workers/Http/HttpFunctionInvocationDispatcher.cs [hunk 2]
```diff
@@ -65,15 +65,14 @@ internal class HttpFunctionInvocationDispatcher : IFunctionInvocationDispatcher
 
         internal Task InitializeHttpWorkerChannelAsync(int attemptCount, CancellationToken cancellationToken = default)
```
Comment (Hazhzeng) [inline src/WebJobs.Script/Workers/Http/HttpFunctionInvocationDispatcher.cs:66]: Does it mean this fix only works for HttpTrigger? Or will it work on other triggers as well?

File: src/WebJobs.Script/Workers/Http/HttpFunctionInvocationDispatcher.cs (hunk 1 above)
Comment (pragnagopa) [reply src/WebJobs.Script/Workers/Http/HttpFunctionInvocationDispatcher.cs]: hmm...deleted commented lines. No new line here.

File: src/WebJobs.Script/Workers/Http/HttpFunctionInvocationDispatcher.cs (hunk 2 above)
Comment (Hazhzeng) [inline src/WebJobs.Script/Workers/Http/HttpFunctionInvocationDispatcher.cs:66]: Thanks for making the change.
I understand the change in `RpcFunctionInvocationDispatcher.cs`. But what is this HttpWorker change for, do we differ the initialization process from other triggers? Or is this change for customer BYO workers?

Comment (pragnagopa) [reply src/WebJobs.Script/Workers/Http/HttpFunctionInvocationDispatcher.cs:66]: yes! this change will also apply to `BYO workers` via [customHandler](https://docs.microsoft.com/en-us/azure/azure-functions/functions-custom-handlers#:~:text=Custom%20handlers%20are%20lightweight%20web,language%20that's%20not%20officially%20supported.)

**Review by Hazhzeng — DISMISSED** [review]
🚢

**Review by mhoeger — APPROVED** [review]
✅ looks great - thanks!!

**Comment by TroyWitthoeft** [comment]
We're currently getting bit by this bug #4384 in production.  We have a python http trigger on consumption plan intermittently failing with: `Did not find any initialized language workers`

~"Anxiously awaiting this fix!  💪  Status?   Where can I track if our our Azure environment gets the latest bits?"~  
Nevermind, found it.  Application Insights (Kusto) has got the sdkversion column which correlate to the release tags.   
I can confirm that we are only seeing this error on older versions, such as 3.0.13901.0   Runs on the latest version are healthy. 
Thanks all.

**Pull request merged**